File: apps/ios/Sources/CoreDomain/Plants/PlantSearchFilterValues.swift
/// The closed vocabularies P11-SEARCH-01's joined plant filters range over.
///
/// `ImageAnalysisKind` is NOT redeclared here — it already exists in
/// `Observations/ObservationPhoto.swift`, and a second copy would be a second
/// thing to keep in step with the same OpenAPI enum.
///
/// Source: packages/api-contracts/openapi.yaml, schemas
/// `TaxonSeasonalActivity`, `PlantDistributionStatus`, and
/// `PlantProfileCompleteness`.

/// The seasonal windows a taxon records. Crop-oriented rather than botanical:
/// these are moments a gardener acts, not the plant's own bloom and fruit
/// phenology, which no current source populates.
public enum TaxonSeasonalActivity: String, Codable, Equatable, Sendable, CaseIterable {
    case sowIndoors = "sow_indoors"
    case sowOutdoors = "sow_outdoors"
    case transplant
    case harvest
}

/// A taxon's standing in a named region. `regulated` is a legal state and
/// `invasive` an ecological one; they overlap often but neither implies the
/// other, so both are asserted independently.
public enum PlantDistributionStatus: String, Codable, Equatable, Sendable, CaseIterable {
    case native
    case introduced
    case invasive
    case regulated
}

/// How much of a taxon's knowledge profile has been materialized. `none` is
/// not a degenerate `partial`: it means enrichment has never produced a
/// profile, the ordinary state for anything the pipeline has not reached.
public enum PlantProfileCompleteness: String, Codable, Equatable, Sendable, CaseIterable {
    case complete
    case partial
    case none
}

File: services/api/src/modules/plants-inventory/application/plant-search-filter-values.ts
/**
 * The closed vocabularies P11-SEARCH-01's joined filters range over.
 *
 * Declared here rather than imported from the modules that own the underlying
 * tables: `image_analysis_result` belongs to observations-history and
 * `plant_distribution_assertion` to integrations, and a search filter reaching
 * into either module's domain to name a value would couple plants-inventory to
 * both for nothing more than four string literals. The values are fixed by
 * database CHECK constraints and by the OpenAPI enums, which is where they are
 * actually enforced; these are the same lists, stated locally.
 *
 * Source: packages/api-contracts/openapi.yaml, schemas `ImageAnalysisKind`,
 * `TaxonSeasonalActivity`, `PlantDistributionStatus`, and
 * `PlantProfileCompleteness`; implementation-plan.md work package
 * P11-SEARCH-01.
 */

export type ImageAnalysisKind = 'stress' | 'disease' | 'pest' | 'other';

export type TaxonSeasonalActivity = 'sow_indoors' | 'sow_outdoors' | 'transplant' | 'harvest';

export type PlantDistributionStatus = 'native' | 'introduced' | 'invasive' | 'regulated';

/** `none` is not a degenerate `partial` — it means enrichment never produced a profile for this taxon at all. */
export type PlantProfileCompleteness = 'complete' | 'partial' | 'none';
